scripts/fbi/hate_crime/preprocess.py

```diff
@@ -0,0 +1,357 @@
+        statvar_list.append(statvar)
+    df_copy['StatVar'] = statvar_dcid_list
+    return df_copy, statvar_list
+
+
+def _write_statvar_mcf(statvar_list, f):
+    dcid_set = set()
+    final_mcf = ''
+    for sv in statvar_list:
+        statvar_mcf_list = []
+        dcid = sv['Node']
+        if dcid in dcid_set:
+            continue
+        dcid_set.add(dcid)
+        for p, v in sv.items():
+            if p != 'Node':
+                statvar_mcf_list.append(f'{p}: dcs:{v}')
+        statvar_mcf = 'Node: dcid:' + dcid + '\n' + '\n'.join(statvar_mcf_list)
+        final_mcf += statvar_mcf + '\n\n'
+
+    f.write(final_mcf)
+
+
+def create_aggr(input_df, config, statvar_list, groupby_cols, agg_dict,
+                population_type):
+    output_df_list = make_time_place_aggregation(input_df,
+                                                 groupby_cols=groupby_cols,
+                                                 agg_dict=agg_dict,
+                                                 multi_index=False)
+
+    for idx in range(len(output_df_list)):
+        output_df_list[idx], statvars = _gen_statvar_mcf(
+            output_df_list[idx], config, population_type=population_type)
+        statvar_list.extend(statvars)
+
+    return output_df_list[1:]  # Skipping national level
+
+
+def _write_to_csv(df, csv_file_name):
+    df['Place'].replace('', np.nan, inplace=True)
+    df.dropna(subset=['Place'], inplace=True)
+    df.to_csv(csv_file_name, index=False)
+
+
+if __name__ == "__main__":
+    df = pd.read_csv('source_data/hate_crime.csv', usecols=INPUT_COLUMNS)
+    fill_col = df.columns[df.isnull().any()].tolist()
+    df[fill_col] = df[fill_col].fillna('Unknown')
+    df['BIAS_AGAINST'] = ''
+    incident_df = df.apply(add_bias_type, axis=1)
+
+    offense_df = flatten_by_column(incident_df, 'OFFENSE_NAME')
+
+    with open('config.json', 'r') as f:
+        config = json.load(f)
+
+    with open('output.mcf', 'w') as f:
+        incident_df, statvar_list = _gen_statvar_mcf(
+            incident_df, config, population_type='CriminalIncidents')
+        _write_statvar_mcf(statvar_list, f)
+
+    final_df = pd.DataFrame()
+
+    # Incidents by StatVar
+    incident_by_statvar = make_time_place_aggregation(
+        incident_df,
+        groupby_cols=['StatVar'],
+        agg_dict={'INCIDENT_ID': 'count'},
+        multi_index=False)
+    final_df = pd.concat(incident_by_statvar)
+    _write_to_csv(final_df, 'output.csv')
+
+    # Aggregations
+
+    # Total Incidents
+    statvar_list = []
+    total_incidents = make_time_place_aggregation(
+        incident_df,
+        groupby_cols=[],
+        agg_dict={'INCIDENT_ID': 'count'},
+        multi_index=False)
+    for idx in range(len(total_incidents)):
+        total_incidents[idx], statvars = _gen_statvar_mcf(
+            total_incidents[idx], config, population_type='CriminalIncidents')
+        statvar_list.extend(statvars)
+
+    final_df = pd.concat(total_incidents[1:])
```

Comment:
this data is not written to csv. can we write it to incidents_count.csv?

Reply:
Thanks for pointing it out.
Have made the fix. Can you take a look if it's ok?